api/client/options.proto

```diff
@@ -121,8 +121,16 @@ message CommandLineOptions {
   }
   // Connection connect timeout period in seconds. Default: 30.
   google.protobuf.Duration timeout = 4 [(validate.rules).duration.gte.seconds = 0];
-  // Use HTTP/2
-  google.protobuf.BoolValue h2 = 5;
+
+  // The protocol to use when encapsulating requests.
+  // Defaults to HTTP/1 if no value is selected.
+  oneof request_protocol {
+    // Use HTTP/2.
+    google.protobuf.BoolValue h2 = 5;
+    // Use HTTP/3 Quic.
+    google.protobuf.BoolValue h3 = 107;
```

Comment:
Could we instead use an enum here (within the oneof)?  If appropriate, we could deprecate the h2 option and have the enum for protocol have http1,2, and 3, and then be more extensible for future protocols as well.

My concern is that as we develop more protocols into nighthawk this approach might become more difficult, especially with the implicit default http1 approach.

(We've discussed this approach online, but posting here for posterity)

Reply:
Done in fbb304d71a7107c03899d2dd175be3fcd3beaaad.